Automattic/wp-calypso
# Plans (State): Remove the Plans (! SitePlans) Redux state completely

Part of addressing https://github.com/Automattic/wp-calypso/issues/86638

## Proposed Changes

* Removes the remaining pieces of Plans state - actions, reducers, constants, etc.
* Removes `QueryPlans` query component
* Updates a few instances of `QueryPlans` with `QueryProducts` to ensure currency-code gets populated for uses of `getCurrentUserCurrenceCode`

## Why are these changes being made?


We only need one form of handling plan pricing in the code base, more so one framework. Anything else risks having alternative and differing views of the same data propagating across the code base.

## Testing Instructions



* Ideally, build passing should suffice
* Otherwise a far more detailed overall testing akin to CfT might also help
* We can ship and handle regressions as they arise. The only ambiguous part I can see is the tricky area with currency-code being partially dependent on plans state

## Pre-merge Checklist
